plugins/dead_code_elimination/dead_code_diff.ml

```diff
@@ -0,0 +1,92 @@
+open Core_kernel
+open Bap.Std
+open Regular.Std
+include Self()
+
+type jmp_change = {
+  cond : exp;
+  kind : jmp_kind;
+} [@@deriving bin_io, sexp, compare]
+
+type change = Rhs of exp | Jmp of jmp_change
+[@@deriving bin_io, sexp, compare]
+
+type diff = Update of change | Kill [@@deriving bin_io, sexp, compare]
+type t = diff Tid.Map.t  [@@deriving bin_io, sexp, compare]
+
+let empty = Tid.Map.empty
+
+let drop_index = (object
+  inherit Exp.mapper
+  method! map_sym var = Var.base var
+end)#map_exp
+
+let changes_in_def d1 d2 =
+  let different = not (phys_equal d1 d2) && Exp.(Def.rhs d1 <> Def.rhs d2) in
+  if different then
+    let d2 = Def.map_exp ~f:drop_index d2 in
+    Some (Term.tid d1, Update (Rhs (Def.rhs d2)))
+  else None
+
+let changes_in_jmp j1 j2 =
+  let different = not (phys_equal j1 j2) && Jmp.(j1 <> j2) in
```

Comment:
do not compare terms directly, this will involve the comparison of all attributes which could be very heavy. Compare only those parts that the algorithm could change.